Zap Logger - Complete Guide to Golang Zap Logger (#500)
SigNoz/signoz.io: docs/tutorial/s3-cold-storage-integration-with-aws-iam-role-eks.md

---

> id: S3-Integration-IAM-ROLE-EKS

Reviewer: ```suggestion
id: s3-integration-iam-role-eks
```

> title: S3 Integration With AWS IAM ROLE IN AWS EKS

Reviewer 2: ```suggestion
title: S3 Integration With AWS IAM role in EKS
```

2. **Using AWS IAM Role.**

To use AWS role to configure S3 cloud storage in signoz we need to add the role arn in the helm **overwrite-values.yaml** and run the helm upgrade.
   ```jsx
       clickhouse:
         cloud: aws
         # Cold storage configuration
         coldStorage:
            enabled: true
            defaultKeepFreeSpaceBytes: "10485760"
            endpoint: https://<AWS S3 DNS URL>/data/
            role:
            enabled: true
            annotations:
               eks.amazonaws.com/role-arn: arn:aws:iam::*********:role/********
   ```
But just adding the AWS role arn will not complete the integration, because here we need to associate the given IAM role with a Kubernetes service account named **clickhouse-instance**. 

### Following are the steps that we need to perform step by step to complete this association of AWS Role ARN with AWS EKS service account named clickhouse-instance.(For this we will configure AWS EKS, OpenID Connect (OIDC) provider, IAM Roles and service accounts.)

1. **Go to EKS cluster and copy the OIDC provider URL.(inside green rectangle)**

> ![AWS OIDC URL](../../static/img/blog/2022/08/aws-eks-oidc-url.png)

Reviewer: Can you use `static/img/docs` folder instead of the blog one for all assets?
You can have a new folder, perhaps `s3-cold-storage` inside `docs` folder.

   ![AWS WEB IDENTITY](../../static/img/blog/2022/08/aws-web-identity.png)

   c. Choose the IAM S3 policy created above and click on the next button.
   ![AWS IAM POLICY](../../static/img/blog/2022/08/aws-iam-policy.png)

   d. Enter the role name **demo-cold-storage-role**, description and tags and create the role.
   ![AWS ADD ROLE](../../static/img/blog/2022/08/aws-add-role.png)

7. **Now update the “overwrite-values.yaml”.**
      ```jsx
         clickhouse:
            cloud: aws
            # Cold storage configuration
            coldStorage:
               enabled: true
               defaultKeepFreeSpaceBytes: "10485760"
               endpoint: https://demo-cold-storage.s3.amazonaws.com/data/
               role:
               enabled: true
               annotations:
                  eks.amazonaws.com/role-arn:arn:aws:iam::1234056789:role/demo-cold-storage-role
      ```
8. **Upgrade the helm deployment.**
   ```jsx
       helm --namespace platform upgrade <your release name> signoz/signoz -f overwrite-values.yaml    
   ```

> # Note: Kinldy change the name of aws resources as per your need.    

Reviewer: Docusaurus should show it as info box

```suggestion
:::info
Kindly change the name of AWS resources as per your need.    
:::
```
